### PR TITLE
[IP-382]: Snapshot company contact data at invoice creation

### LINKED ISSUE #382: [Feature Request] Possibility to change own contact data without changing existing invoices
i just move to another address and realized that there seems no way to change my contact and phone data without having the effect that all old invoices now have the same new contact data.

a feature that stores changed contact data separately in db and map that field id against each invoice would be awesome.

### PR DESCRIPTION
**Pull request by KhawarMehfooz**
# Pull Request Checklist   

## Checklist  
- [x] My code follows the code formatting guidelines.  
- [x] I have tested my changes locally.  
- [x] I selected the appropriate branch for this PR.  
- [x] I have rebased my changes on top of the selected branch.  
- [ ] I included relevant documentation updates if necessary.  
- [x] I have an accompanying issue ID for this pull request.  

---

## Description  

Invoices now snapshot the issuing company's name, VAT number, ID number, and Chamber-of-Commerce number at the moment they're created, instead of always reading them live off the `companies` table. The invoice PDF/preview template, the credit note flow, invoice duplication, and the "Email Invoice" placeholder all now prefer this frozen snapshot and fall back to the live company record for invoices created before this change.

- Added a migration adding `company_name`, `company_vat_number`, `company_id_number`, `company_coc_number` (nullable) to `invoices`.
- `InvoiceService::createInvoice()` populates the snapshot from the current company at creation time.
- `InvoiceService::createCreditNote()` copies the snapshot from the parent invoice (not the live company), so a credit note stays consistent with the invoice it corrects.
- `InvoiceCopyService::copy()` (duplicate invoice) snapshots the *current* company, since a duplicate is effectively a new invoice.
- `pdf/invoice.blade.php` renders `company_name`/`company_vat_number` with `?? $invoice->company?->name` fallback for pre-existing invoices, and now also displays the company's VAT number (previously not shown at all).
- Fixed the same "renamed company retroactively changes old data" bug in the invoice email template's `company.name` placeholder.
- Added `Modules/Invoices/Tests/Feature/InvoiceCompanySnapshotTest.php` covering: snapshot captured at creation, a later company rename doesn't change an existing invoice, the PDF renders the frozen name not the live one, pre-migration invoices fall back correctly, credit notes inherit the parent's snapshot, and duplicated invoices pick up the current company.

**Scope note:** this covers the company fields that are actually editable in the app today (name, VAT/ID/COC numbers via the admin Company form and the company-panel Settings page). Company address and phone/email have no editing UI anywhere in the app yet — building that, and snapshotting it, is a separate follow-up.

---

## Related Issue(s)  

Fixes #382

---

## Motivation and Context  

A company that updates its details (e.g. after moving address, or a legal rename) had every past invoice's PDF silently change to show the new name, since the invoice header was always rendered live from the `Company` relationship rather than the data that was true when the invoice was issued. This is a real compliance problem — old invoices should stay legally accurate to what was true at the time they were sent. This PR applies the same denormalized-snapshot pattern the codebase already uses for invoice line items (`invoice_items` stores product name/price at time of sale) to the invoice header's company details.

---

## Issue Type (Check one or more)  
- [x] Bugfix  
- [ ] Improvement of an existing feature  
- [ ] New feature  

---

## Screenshots  
N/A

---

## Summary by CodeRabbit

* **New Features**
  * Invoices now preserve the company name and registration details from the time they are issued.
  * Invoice previews and PDFs retain original company information even after company details change.
  * Credit notes and copied invoices inherit the relevant company details.
  * Email placeholders use the preserved invoice company name when available.

* **Bug Fixes**
  * Improved handling of missing company records or incomplete company details.

### CODE REVIEW

**Comment by nielsdrost7** [comment]
@KhawarMehfooz 178 commits?
What's going on? Weird rebase or something?

Luckily only 6 changed files

File: Modules/Invoices/Tests/Unit/InvoiceCompanySnapshotTest.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        $this->company->update([
+            'name'        => 'ACME Corp',
+            'vat_number'  => 'BE0123456789',
+            'id_number'   => 'ID-1',
+            'coc_number'  => 'COC-1',
+        ]);
+        $customer = Relation::factory()->for($this->company)->customer()->create();
+
+        /* Act */
+        $invoice = app(InvoiceService::class)->createInvoice($this->invoicePayload($customer));
```
Comment (nielsdrost7) [inline Modules/Invoices/Tests/Unit/InvoiceCompanySnapshotTest.php:37]: @KhawarMehfooz can you move this to a Unit test? We're basically testing the InvoiceService Unit

**Comment by KhawarMehfooz** [comment]
@nielsdrost7 
I'd been syncing my fork's develop with git merge while upstream squash-merges PRs, so it accumulated 177 duplicate-content commits under different SHAs. I have rebased my branch straight onto upstream/develop and it's down to 1 commit now.

Learning new things... :)

**Comment by nielsdrost7** [comment]
@KhawarMehfooz tiny, tiny review is on its way, stand by

**Comment by KhawarMehfooz** [comment]
@nielsdrost7 No problem. take your time.

**Review by nielsdrost7 — CHANGES_REQUESTED** [review]
@KhawarMehfooz i saw 1 edge case that might need a phpunit test:

> **Edge cases worth covering**: client with no address set at all (nullable snapshot fields), and a client that's soft-deleted after the invoice exists (invoice must still render its snapshot, not fail on a missing relation).

And also I think
- [ ] actually saving an invoice through Filament (that's the Feature test)
- [ ] changing addresses on the Company Model
- [ ] and then at the end Asserting that the old address is still shown on the Invoice Preview? You think that's possible?

Todo:
- [ ] Move Snapshot test class to Unit tests

File: Modules/Invoices/Tests/Unit/InvoiceCompanySnapshotTest.php (hunk 1 above)
Comment (KhawarMehfooz) [reply Modules/Invoices/Tests/Unit/InvoiceCompanySnapshotTest.php:37]: @nielsdrost7 On it.

**Comment by nielsdrost7** [comment]
@KhawarMehfooz thanks man!
Merged
